src/config.c

```diff
@@ -220,6 +267,61 @@ nm_config_t *nm_config_parse(char **err_out) {
     #undef NM_ERR_RET
 }
 
+void nm_config_generate(nm_config_t *cfg) {
+    NM_LOG("config: removing any previously generated items");
+    for (nm_config_t *prev = NULL, *cur = cfg; cur; cur = cur->next) {
+        if (prev && cur->generated) { // we can't get rid of the first item and it won't be generated anyways (plus doing it this way simplifies the loop)
+            prev->next = cur->next; // skip the generated item
+            cur->next = NULL; // so we only free that item
+            nm_config_free(cur);
+            cur = prev; // continue with the new current item
+        } else {
+            prev = cur; // the item was kept, so it's now the previous one
+        }
+    }
+
+    NM_LOG("config: running generators");
+    for (nm_config_t *cur = cfg; cur; cur = cur->next) {
+        if (cur->type == NM_CONFIG_TYPE_GENERATOR) {
+            NM_LOG("config: running generator %s:%s", cur->value.generator->desc, cur->value.generator->arg);
+
+            size_t sz;
+            nm_menu_item_t **items = nm_generator_do(cur->value.generator, &sz);
+            if (!items) {
+                NM_LOG("config: ... no items generated");
+                continue;
+            }
+
+            NM_LOG("config: ... %zu items generated", sz);
+            for (size_t i = 0; i < sz; i++) {
+                nm_config_t *tmp = calloc(1, sizeof(nm_config_t));
+                tmp->type = NM_CONFIG_TYPE_MENU_ITEM;
+                tmp->value.menu_item = items[i];
+                tmp->generated = true;
+                tmp->next = cur->next;
+                cur->next = tmp;
+            }
+            free(items);
+        }
+    }
+
+    NM_LOG("config: generated items");
+    for (nm_config_t *cur = cfg; cur; cur = cur->next) {
+        if (cur->generated) {
+            switch (cur->type) {
+            case NM_CONFIG_TYPE_MENU_ITEM:
+                NM_LOG("cfg(NM_CONFIG_TYPE_MENU_ITEM) : %d:%s", cur->value.menu_item->loc, cur->value.menu_item->lbl);
+                for (nm_menu_action_t *cur_act = cur->value.menu_item->action; cur_act; cur_act = cur_act->next)
+                    NM_LOG("...cfg(NM_CONFIG_TYPE_MENU_ITEM) (%s%s%s) : %p:%s", cur_act->on_success ? "on_success" : "", (cur_act->on_success && cur_act->on_failure) ? ", " : "", cur_act->on_failure ? "on_failure" : "", cur_act->act, cur_act->arg);
+                break;
+            case NM_CONFIG_TYPE_GENERATOR:
```

Comment:
Is the intent behind this branch being able to generate generators, somehow?

Or is that a leftover from a c/p from `nm_config_parse` above?

(Or did I miss something? ^^. Right now, I don't think this branch can ever be reached?)

Reply:
The first few versions of this feature generated `nm_config_t`s. which meant they could recursively generate generators. I ultimately got rid of this feature, as I felt it would make `generator` too powerful, make them dependent on implementation details of the config parser (which would need to be exposed in the header file) and reducing the clear separation between files, provide more opportunity for mistakes (recursion of generators), and would be useless in the majority of cases (or there would be something else more suitable/easy to understand for a user).

I'll leave that branch in for completeness and copy-pasteability.